modules/javafx.controls/src/main/java/javafx/scene/control/skin/VirtualFlow.java

```diff
@@ -2865,6 +2936,110 @@ private double computeOffsetForCell(int itemIndex) {
         return -(getViewportLength() * p);
     }
 
+    double getCellSize(int idx) {
+        return getOrCreateCellSize(idx, false);
+    }
+
+    /**
+     * Get the size of the considered element.
+     * If the requested element has a size that is not yet in the cache,
+     * it will be computed and cached now.
+     * @return the size of the element; or 1 in case there are no cells yet
+     */
+    double getOrCreateCellSize(int idx) {
+        return getOrCreateCellSize (idx, true);
+    }
+
+    private double getOrCreateCellSize (int idx, boolean create) {
+        // is the current cache long enough to contain idx?
+        if (itemSizeCache.size() > idx) {
+            // is there a non-null value stored in the cache?
+            if (itemSizeCache.get(idx) != null) {
+                return itemSizeCache.get(idx);
+            }
+        }
+        if (!create) return -1;
+        boolean doRelease = false;
+
+        // Do we have a visible cell for this index?
+        T cell = getVisibleCell(idx);
+        if (cell == null) { // we might get the accumcell here
+            cell = getCell(idx);
+            doRelease = true;
+        }
+        // Make sure we have enough space in the cache to store this index
+        while (idx >= itemSizeCache.size()) {
+            itemSizeCache.add(itemSizeCache.size(), null);
```

Comment:
Can this be simplified to `itemSizeCache.add(null);`?

Reply:
That won't add enough elements. For example, in case idx = 10, and the itemSizeCache has 5 elements, we need to add 5 empty elements to the cache, as they might get queried/pushed, e.g. we might do
`itemSizeCache.set(7, 123.45);`

Reply:
Sure, that's why you need the `while` loop. What I meant is that the following two calls are equivalent for any list:

```
    list.add(list.size(), null);
```

and

```
    list.add(null);
```